### PR TITLE
refactor(accessapproval): versioned clients

### PR DESCRIPTION
**Pull request by dbolduc**
Part of the work for #10170 

Move `accessapproval`, leaving behind backwards compatibility forwarding headers.

It turns out that `check-api` fails. I am not sure why. I think it just doesn't have member functions when we say `using SomeClass;`. Googlers can see the generated [report](https://paste.googleplex.com/6094068080705536).

## Procedure

This can act as a prototype for moving other libraries. I do not have it down to a science, but the steps are roughly:

1. Setup
```sh
library=... # e.g. accessapproval
git checkout -b relocate-$library
```

2. Update generator config (e.g. `product_path: google/cloud/foo/v1`; `forwarding_product_path: google/cloud/foo`)
```sh
edit generator/generator_config.textproto
git commit -am "refactor($library): versioned clients"
```

3. Regenerate
```sh
ci/cloudbuild/build.sh -t generate-libraries-pr
ci/cloudbuild/build.sh -t checkers-pr
git commit -am "generated changes"
```

4. Cleanup stale files
```sh
rm google/cloud/$library/*.cc
rm -rf google/cloud/$library/internal
rm -rf google/cloud/$library/samples
```

5. Manual edits to build files.
```sh
edit google/cloud/$library/BUILD.bazel
edit google/cloud/$library/CMakeLists.txt
```

6. Test
```sh
ci/cloudbuild/build.sh -t cmake-install-pr
bazel build //google/cloud/$library/...
```

7. Format, commit
```sh
ci/cloudbuild/build.sh -t checkers-pr
git commit -am "manually update builds; remove stale code"
```

8. Update ABI dump
```sh
ci/cloudbuild/build.sh -t check-api-pr
git commit -m "abi update" ci/abi-dumps/google_cloud_cpp_$library.expected.abi.dump.gz
```

Then **after the versioned clients have been released and included in vcpkg**, update the quickstart to use the versioned client.
```sh
git checkout -b $library-quickstart-uses-versioned-client
edit google/cloud/$library/quickstart/quickstart.cc
ci/cloudbuild/build.sh -t checkers-pr
git commit -am "cleanup($library): quickstart uses versioned client"
```

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10463)

### CODE REVIEW

File: google/cloud/accessapproval/access_approval_client.h [hunk 1]
```diff
@@ -20,440 +20,14 @@
...
-  GetAccessApprovalServiceAccount(
-      google::cloud::accessapproval::v1::
-          GetAccessApprovalServiceAccountMessage const& request,
-      Options opts = {});
-
- private:
-  std::shared_ptr<AccessApprovalConnection> connection_;
-  Options options_;
-};
+using ::google::cloud::accessapproval_v1::AccessApprovalClient;
```
Comment (devbww) [inline google/cloud/accessapproval/access_approval_client.h:30]: Should the forwarding headers say anything about preferring the versioned names, like an `@deprecated` or an `@note`?  (Sorry I didn't ask this in the generator PR.)

Comment (dbolduc) [reply google/cloud/accessapproval/access_approval_client.h:30]: I think so. I opened #10466. I'll rebase this one (and probably clean up the quickstart commits) after it goes through.

Comment (devbww) [reply google/cloud/accessapproval/access_approval_client.h:30]: SGTM.

**Review by devbww — APPROVED** [review]

**Pull request merged**